src/efi.rs

```diff
@@ -33,16 +35,47 @@ impl EFI {
         Path::new(MOUNT_PATH).join("EFI")
     }
 
+    fn esp_device(&self) -> PathBuf {
+        Path::new("/dev/disk/by-partlabel/").join(ESP_PART_LABEL)
```

Comment:
This still assumes there'll be a canonical ESP.  :slightly_smiling_face: 

RAIDed boot disks won't even have a partition with this label.  Instead we should use libblkid or similar to walk any filesystem with the partition GUID `c12a7328-f81f-11d2-ba4b-00a0c93ec93b`.

Reply:
I thought there is still a lot of work to be done related to supporting redundant disks for `/boot` and `/boot/efi`. I was thinking this PR would at least let bootupd support the case where we're not using redundant disks, but the EFI is also not mounted (please correct me if I'm wrong!).

Reply:
Redundant disks for `/boot` should be transparent to bootupd.  But for `/boot/efi`, you're right, and yes, this PR should be sufficient for the non-redundant case.  Let's update the commit message to reflect that goal, and also make sure we have a good error message when we encounter an unsupported environment.

Reply:
ah ok, that makes sense! Thanks for the explanation :)

Reply:
Opened https://github.com/coreos/bootupd/issues/132 to track this.